socketcan_bridge/src/socketcan_to_topic_nodelet.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Enway GmbH - All Rights reserved.
+ * Proprietary & confidential.
+ */
+
+#include "socketcan_bridge/socketcan_to_topic_nodelet.h"
+
+#include <pluginlib/class_list_macros.h>
+
+PLUGINLIB_EXPORT_CLASS(socketcan_bridge::SocketcanToTopicNodelet, nodelet::Nodelet)
+
+socketcan_bridge::SocketcanToTopicNodelet::~SocketcanToTopicNodelet()
+{
+  driver_->shutdown();
+  driver_.reset();
+}
+
+void socketcan_bridge::SocketcanToTopicNodelet::onInit()
+{
+  ros::NodeHandle nh = getPrivateNodeHandle();
+  std::string can_device;
+  nh.param<std::string>("can_device", can_device, "can0");
+
+  driver_ = boost::make_shared<can::ThreadedSocketCANInterface> ();
+  if (!driver_->init(can_device, 0))  // initialize device at can_device, 0 for no loopback.
+  {
+    NODELET_FATAL("Failed to initialize can_device at %s", can_device.c_str());
+    return;
+  }
+    else
+  {
+    NODELET_INFO("Successfully connected to %s.", can_device.c_str());
+  }
+
+  to_topic_bridge_ptr_.reset(new socketcan_bridge::SocketCANToTopic(&nh, &nh, driver_));
```

Comment:
It looks weird to pass the same handle twice, isn't one of them maybe supposed to be private and the other not?

Reply:
Also please use std::make_unique instead

Reply:
for the node handle, my understanding is that we always use the private node handle. so that we always resolve param/topic relative to the node namespace

Reply:
I think that makes sense for advertised topics and services, but in my opinion subscriptions should be in public space. If however this is just the same as the existing node of the socketcan_bridge works, then it should be the same. So I guess it's okay.

Reply:
I did spend some more thoughts on this. I feel private handle works better for nodelet. partly because socketcan_bridge has the pub/sub topic hardcoded as "received_messages" / "sent_messages".
The node version uses public nodehandle for both publishing and subscribing to toipcs. We have to give prefix to nodes in order to have multiple can interfaces, without remapping them explicitly. 
For nodelet, if we want different can device to share the same nodelet manager, then public handle for these nodelet will all be mapped to same topic "/manager/received_messages"  or "/manager/sent_messages"
That's why I feel it makes more sense to put in the nodelet name. so we can have something like "manager/can1/received_messages" or "/manager/can1/sent_messages"

let me know your thoughts.

Reply:
I would prefer explicit remapping because we keep the same behavior as the node version (minimize surprises). In principle I would like to keep the published topic in private namespace (a convention [we have](https://enway-gmbh-cleansquare-ros.readthedocs-hosted.com/en/latest/dev_workflow.html#general-practices-hints)) but I think keeping the same behavior as the node dominates here.

Reply:
Ok, agree. I will align the usage of node-handle with the node version. Will do explicit remapping to address the overlapping of topics.